lerobot/common/policies/diffusion/modeling_diffusion.py

```diff
@@ -749,3 +768,264 @@ def forward(self, x: Tensor, cond: Tensor) -> Tensor:
+        if config.n_cond_layers > 0:
+            encoder_layer = nn.TransformerEncoderLayer(
+                d_model=config.diffusion_step_embed_dim,
+                nhead=config.n_head,
+                dim_feedforward=4 * config.diffusion_step_embed_dim,
+                dropout=config.p_drop_attn,
+                activation="gelu",
+                batch_first=True,
+                norm_first=True,
+            )
+            self.encoder = nn.TransformerEncoder(encoder_layer=encoder_layer, num_layers=config.n_cond_layers)
+        else:
+            self.encoder = nn.Sequential(
+                nn.Linear(config.diffusion_step_embed_dim, 4 * config.diffusion_step_embed_dim),
+                nn.Mish(),
+                nn.Linear(4 * config.diffusion_step_embed_dim, config.diffusion_step_embed_dim),
+            )
+        # decoder
+        decoder_layer = nn.TransformerDecoderLayer(
+            d_model=config.diffusion_step_embed_dim,
+            nhead=config.n_head,
+            dim_feedforward=4 * config.diffusion_step_embed_dim,
+            dropout=config.p_drop_attn,
+            activation="gelu",
+            batch_first=True,
+            norm_first=True,  # important for stability
+        )
+        self.decoder = nn.TransformerDecoder(decoder_layer=decoder_layer, num_layers=config.n_layer)
+
+        # attention mask
+        if config.causal_attn:
+            # causal mask to ensure that attention is only applied to the left in the input sequence
+            # torch.nn.Transformer uses additive mask as opposed to multiplicative mask in minGPT
+            # therefore, the upper triangle should be -inf and others (including diag) should be 0.
+            sz = t
+            mask = (torch.triu(torch.ones(sz, sz)) == 1).transpose(0, 1)
+            mask = mask.float().masked_fill(mask == 0, float("-inf")).masked_fill(mask == 1, float(0.0))
+            self.register_buffer("mask", mask)
+
+            # assume conditioning over time and observation both
+            p, q = torch.meshgrid(torch.arange(t), torch.arange(t_cond), indexing="ij")
+            mask = p >= (q - 1)  # add one dimension since time is the first token in cond
+            mask = mask.float().masked_fill(mask == 0, float("-inf")).masked_fill(mask == 1, float(0.0))
+            self.register_buffer("memory_mask", mask)
+        else:
+            self.mask = None
+            self.memory_mask = None
+
+        # decoder head
+        self.ln_f = nn.LayerNorm(config.diffusion_step_embed_dim)
+        self.head = nn.Linear(config.diffusion_step_embed_dim, input_dim)
+
+        # constants
+        self.t = t
+        self.t_cond = t_cond
+        self.horizon = config.horizon
+        self.n_obs_steps = config.n_obs_steps
+
+        # init
+        self.apply(self._init_weights)
+        # logger.info(
+        #     "number of parameters: %e", sum(p.numel() for p in self.parameters())
+        # )
+
+    def _init_weights(self, module):
```

Comment:
Is there anything we can say in a docstring to summarize the weight initialization strategy used here?